server/app/routes/user/index.js

```diff
@@ -29,19 +31,24 @@ router.param('id', (req, res, next, id) => {
 });
 
 router.put("/:id", (req, res, next) => {
-	User.findById(req.requestUser._id)
-	.then(user => {
-		user.set(req.body);
-		return user.save();
-	})
-	.then(updatedUser => res.json(updatedUser))
-	.catch(next);
+	if(!req.user) res.sendStatus(401)
+	else if (req.user.isAdmin || req.user.equals(req.requestUser)){
+		User.findById(req.requestUser._id)
+		.then(user => {
+			user.set(req.body);
+			return user.save();
+		})
+		.then(updatedUser => res.json(updatedUser))
+		.catch(next);
+	} else res.sendStatus(401)
 });
 
 router.delete('/:id', (req, res, next) => {
-	req.requestUser.remove()
-	.then(() => res.sendStatus(204))
-	.catch(next);
+	if(req.user || req.user.isAdmin){
```

Comment:
code is slightly different for update user.  there you have req.user.equals(req.requestUser) -- should this be like that also?